src/main/java/com/celements/navigation/cmd/GetMappedMenuItemsForParentCommand.java

```diff
@@ -64,28 +71,42 @@ public boolean is_isActive() {
       List<TreeNode> menu = new ArrayList<TreeNode>();
       try {
         List<Object[]> results = context.getWiki().getStore().search(getHQL(), 0, 0, context);
-        mLogger.info("getMenuItems: found " + results.size() + " menus with parentKey "
-            + parentKey);
+        LOGGER.info("getMenuItems: found " + results.size() + " menus with parentKey " + parentKey);
         for (Object[] nodeData : results) {
-          TreeNode treeNode = new TreeNode(new DocumentReference(context.getDatabase(),
-              nodeData[1].toString(), nodeData[0].toString().split("\\.")[1]),
-              nodeData[2].toString(), (Integer) nodeData[3]);
-          treeNode.setPartName(nodeData[4].toString());
+          DocumentReference docRef = new DocumentReference(context.getDatabase(),
+              nodeData[1].toString(), nodeData[0].toString().split("\\.")[1]);
+          String partName = MoreObjects.firstNonNull(nodeData[4], "").toString();
+          String parentFN = MoreObjects.firstNonNull(nodeData[2], "").toString();
+          TreeNode treeNode = new TreeNode(docRef, resolveParentRef(parentFN),
+              (Integer) nodeData[3], partName);
           menu.add(treeNode);
         }
       } catch (XWikiException e) {
-        mLogger.error("getMenuItems ", e);
+        LOGGER.error("getMenuItems ", e);
       }
       context.setDatabase(saveDatabase);
       return menu;
     } else {
       if (is_isActive()) {
-        mLogger.warn("getMenuItems: parentKey is emtpy");
+        LOGGER.warn("getMenuItems: parentKey is emtpy");
       }
       return Collections.emptyList();
     }
   }
 
+  private EntityReference resolveParentRef(@NotNull String parentFN) {
+    EntityType entityType = (parentFN.contains(".")) ? EntityType.DOCUMENT : EntityType.SPACE;
+    EntityReference parentRef = null;
+    if (parentFN.isEmpty()) {
```

Comment:
I'm guessing a NOT operator to be missing here. Please add unit tests for this method.